src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1311 @@
+        for i in range(self.rank()):
+            cvar_poly_pairs.append(latex(tuple([cvars[i], polys[i]])))
+
+        return latex(tuple(cvar_poly_pairs))
+
+    def are_laurent_polys_trivial(self):
+        r"""
+        Returns whether this seed has Laurent polynomials with nontrivial denominators.
+        """
+        for poly in self.laurent_polys():
+
+            if poly.denominator() != 1:
+
+                return False
+
+        return True
+
+    def is_mutation_infinite(self, give_reason=True):
+        r"""
+        Performs some heuristic checks on the mutation class of this seed to work out if it is mutation infinite or not.
+        Completely probabilistic and may false positive; do not use in research setting.
+        Warning: This could take a long time for some seeds!
+        """
+
+        MAX_DEGREE = 20  # when to decide a seed mutation class is too large based on degrees of exchange polynomials
+        counter = 0
+        for seed in self.mutation_class_iter():
+            counter += 1
+            if not seed.passes_rank_two_check():  # type: ignore
+                if give_reason:
+                    print("Fails rank two check after mutating at indices %s" %
+                          (seed.mutation_sequence()))
+                return True
+
+            # check degrees of exchange polynomials
+
+            for poly in seed.exchange_polys():  # type: ignore
+
+                for var in poly.variables():
+
+                    if poly.degree(var) > MAX_DEGREE:
+                        if give_reason:
+                            print("Fails polynomial degree finiteness check after mutating at indices %s" % (
+                                seed.mutation_sequence()))
+                        return True
+
+        # if we managed to iterate through everything, then it must be finite
+
+        if give_reason:
+            print("Mutation finite, has %s seeds in mutation class" % (counter))
+        return False
+
+    def passes_rank_two_check(self):
+        r"""
+        Check that, when restricting to any two variables, that the product of the degrees is strictly less than 4.
+        """
+
+        N = self.rank()
+        polys = self.exchange_polys()
+
+        for i in range(N):
+            for j in range(i, N):
+                if polys[i].degree(self._polynomial_ring(self._names[j])) * polys[j].degree(self._polynomial_ring(self._names[i])) >= 4:
+                    return False
+
+        return True
+
+    def get_laurent_poly_denominators(self):
+        r"""
+        Returns a list containing the denominators of the Laurent polynomials for this seed.
+        """
+        return [poly.denominator() for poly in self.laurent_polys()]
+
+    @staticmethod
+    def create_generic_seed(vars, polys):
+
+        for i in range(len(polys)):
+
+            coefficients = []
+            M = polys[i].monomials()
+            new_coeffs = list(var('a_%d%d' % (i, j) for j in range(len(M))))
```

Comment:
```suggestion
            new_coeffs = list(SR.var('a_%d%d' % (i, j) for j in range(len(M))))
```
This is a bit better for constructing symbolic variables. Although you might want to reconsider the variable names for cases when there are 10 variables or more.